sktime/transformations/series/fourier.py

```diff
@@ -149,9 +166,21 @@ def _fit(self, X, y=None):
                         "exists from other seasonal period, fourier term pairs."
                     )
 
-        # store the integer form of the minimum date in the prediod index
         # this is used to make sure that time t is calculated with reference to
         # the data passed on fit
+        if isinstance(X.index, pd.DatetimeIndex):
+            # Chooses first non None value
+            freq = X.index.freq or self.freq or pd.infer_freq(X.index)
```

Comment:
I think we need to store this `freq` as an attribute, maybe `self.freq_` and use this instead in the predict step.